src/test/java/org/hydev/mcpm/client/models/PluginYmlTest.java

```diff
@@ -0,0 +1,69 @@
+package org.hydev.mcpm.client.models;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.Test;
+
+class PluginYmlTest
+{
+    @Test
+    void fromYml() throws PluginYml.InvalidPluginMetaStructure, JsonProcessingException
+    {
+        String yml = """  
+            main: org.hydev.mcpm.SpigotEntry
+            name: MCPM
+            version: 1.1
+            description: Minecraft Package Manager for Bukkit/Spiget Servers
+            api-version: 1.18
+            authors: Hykilpikonna
+            website: https://github.com/CSC207-2022F-UofT/mcpm
+            commands:
+              mcpm:
+                description: Minecraft package manager
+            """.stripIndent();
+
+        var meta = PluginYml.fromYml(yml);
+
+        assert meta.main().equals("org.hydev.mcpm.SpigotEntry");
+        assert meta.name().equals("MCPM");
```

Comment:
This is good. It makes me question if it would be sane for PluginYml to override equals.

Reply:
it does... records does override equals but yea making a PluginYml and calling equals would be better

Reply:
Like deep equals? That's pretty cool!